src/zebra.c

```diff
@@ -3556,21 +3556,33 @@ static void draw_zoom_overlay(int dirty)
 
 int liveview_display_idle()
 {
+#ifdef CONFIG_DIGIC_678
+    extern struct dialog* LiveViewApp_dialog;
+
+/**
+ * kitor TODO: contains a lot of duplicates from D45 code.
+ * Do we want to keep it that way? Seems cleaner to me.
+ */
+    return
+        LV_NON_PAUSED &&
+        DISPLAY_IS_ON &&
+        !menu_active_and_not_hidden() &&
+        LiveViewApp_dialog &&
+        gui_state == GUISTATE_IDLE &&
+        CURRENT_GUI_MODE <= 3 &&
+        job_state_ready_to_take_pic() &&
+        !mirror_down;
+#else
     struct gui_task * current = gui_task_list.current;
     struct dialog * dialog = current->priv;
     extern thunk LiveViewApp_handler;
 
     #if defined(CONFIG_5D3) || defined(CONFIG_M50)
     extern thunk LiveViewLevelApp_handler;
```

Comment:
Changes below in that function just restore code prior to initial D6+ modifications.